rpc/wrtc_client.go

```diff
@@ -303,7 +314,6 @@ func dialWebRTC(
 	clientCh := newWebRTCClientChannel(peerConn, dataChannel, onICEConnected, logger, dOpts.unaryInterceptor, dOpts.streamInterceptor)
 
 	exchangeCandidates := func() error {
-		haveInit := false
```

Comment:
I can't comment off-diff, but [I think this "swallow EOF"s isn't right](https://github.com/viamrobotics/goutils/blob/main/rpc/wrtc_client.go#L315-L321). I think that should be:
```
			callResp, err := callClient.Recv()
			if err != nil {
				if !haveInit && !errors.Is(err, io.EOF) {
					return err
				}
				return nil
			}
```

Reply:
Can we just remove the swallowing from that section and let `sendErr` handle the swallowing?

Reply:
yeah we can just return err and the goroutine that calls `exchangeCandidates()` will `sendErr`